test/allow-extended-operators.test.js

```diff
@@ -47,16 +32,212 @@ describe('Model.settings.allowExtendedOperators', () => {
     return new DataSource(connectorSettings);
   }
 
+  function setCustomData() {
+    return {$set: {value: 'changed'}};
+  }
+
   class TestConnector {
     constructor(dataSource) {
     }
 
+    create(model, data, options, callback) {
+      callback();
+    }
+
+    updateAttributes(model, id, data, options, callback) {
+      callback();
+    }
+
     all(model, filter, options, callback) {
       // return the raw "value" query
-      var instanceFound = {
+      let instanceFound = {
         value: filter.where.value,
       };
       callback(null, [instanceFound]);
     }
   }
+
+  describe('connector.settings.allowExtendedOperators', () => {
+    context('DAO.find()', () => {
+      it('converts extended operators to string value by default', () => {
+        const TestModel = createTestModel();
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql('[object Object]');
+        });
+      });
+
+      it('preserves extended operators with allowExtendedOperators set', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true});
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql({$exists: true});
+        });
+      });
+
+      function extendedQuery() {
+        // datasource modifies the query,
+        // we have to build a new object for each test
+        return {where: {value: {$exists: true}}};
+      }
+    });
+
+    context('DAO.updateAttributes()', () => {
+      it('`options.allowExtendedOperators` override connector settings - disable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: false}, {strict: true});
+        return TestModel.create({value: 'test'}).then((instance) => {
+          return instance.updateAttributes(setCustomData(), {allowExtendedOperators: true})
+            .then(() => {
+              should(TestModel.lastPersistedData).eql(setCustomData());
+            });
+        });
+      });
+
+      it('`options.allowExtendedOperators` override connector settings - enable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true}, {strict: true});
+        return TestModel.create({value: 'test'}).then((inst) => {
+          return inst.updateAttributes(setCustomData(), {allowExtendedOperators: false})
+            .then(() => {
+              throw new Error('updateAttributes() should have failed.');
+            })
+            .catch(function(err) {
```

Comment:
This will catch the error produced two lines above and then fail on assertion checking that the error was `ValidationError`, hiding the fact that `updateAttributes` actually succeeded.

It's important to pass the assertion function as the second argument of `then()` instead. If you are using regular functions like here, then please name them (we typically use `onSuccess` and `onError`).